paimon-core/src/main/java/org/apache/paimon/casting/CastExecutors.java

```diff
@@ -97,6 +97,8 @@ public class CastExecutors {
                                         }
                                 }
                             };
+                        case VARCHAR:
+                            return Object::toString;
```

Comment:
Return type should be `BinaryString`, not `String`. See the following case branches about `CHAR` and `VARCHAR`.

Reply:
done.